tests/queries/0_stateless/03202_system_load_primary_key.sql

```diff
@@ -0,0 +1,11 @@
+-- Test to ensure SYSTEM LOAD PRIMARY KEY works as expected
+CREATE TABLE test_load_primary_key (id Int32, value String) ENGINE = MergeTree() ORDER BY id;
+
+-- Inserting some data
+INSERT INTO test_load_primary_key VALUES (1, 'a'), (2, 'b'), (3, 'c');
+
+-- Load primary key
+SYSTEM LOAD PRIMARY KEY test_load_primary_key;
+
+-- Select to verify the data is correctly loaded
+SELECT * FROM test_load_primary_key ORDER BY id;
```

Comment:
In case l. 8 did _not_ load the primary key (because of a potential bug), l. 11 will load it on the fly. Therefore, the test does not fully fulfill its mission (l. 1). 

Could you rewrite the tests such that it checks the memory size of the index similar to https://github.com/ClickHouse/ClickHouse/pull/62738/files#diff-dc0e5a27310472c030dbe3f584822f28a4daf74897c98822cc697d985bbe225f ? Please also add tag "no-parallel" to avoid multiple instances of this test running concurrently (which would mess with the tests setup).

Reply:
+1, please check against `system.parts` to make sure whether the index is loaded or not after every assumption. In order to unload it, you may want to use `SYSTEM UNLOAD PRIMARY KEY`

Reply:
Now, the test explicitly checks the memory size of the primary key in system.parts both before and after loading, ensuring that the primary key is not loaded on the fly. Also "no-parallel" tag was added to prevent any interference. Also SYSTEM UNLOAD PRIMARY KEY to ensure that all primary keys were unloaded afterward

Reply:
I still see the old code

Reply:
New changes have been pushed!

Reply:
There are 2 issues I find here:

1. You're only testing that unloading a primary key works for a particular table. Shouldn't we ensure it works also for all tables? Notice how there's a `03127_system_unload_primary_key_table` and then a `03128_system_unload_primary_key` for the `SYSTEM LOAD PRIMARY KEY` counterpart
2. Even though you're expecting different results to distinguish when the primary index has actually been loaded, the reference always shows that it's never loaded.

Reply:
Thanks for the feedback! I’ve expanded the test to cover multiple tables and updated it to ensure the primary key memory is correctly checked before and after loading/unloading. The reference file now reflects the actual memory usage changes.

Reply:
Much better! But we're still missing testing loading all primary keys. In other words, with the current test we're not testing the path where multiple tasks are scheduled in the thread pool.

If you don't want to separate loading primary keys for one table in one test and for all tables in a separate test, you could still do something as follows to check both paths. It's just a suggestion, feel free to come up with something else as long as it tests all paths:

1. Create 2 tables :white_check_mark: 
2. Ensure their keys are not loaded :white_check_mark: 
3. Load them all using `SYSTEM LOAD PRIMARY KEY`
4. Ensure their keys are loaded
5. Unload them all using `SYSTEM UNLOAD PRIMARY KEY`
6. Ensure keys are unloaded
7. Load only one of the two tables using `SYSTEM LOAD PRIMARY KEY table_name`
8. Ensure only one key is actually loaded